Add 2.4.0 release notes (#5923)
elastic/cloud-on-k8s: docs/release-notes/2.4.0.asciidoc

:issue: https://github.com/elastic/cloud-on-k8s/issues/
:pull: https://github.com/elastic/cloud-on-k8s/pull/

[[release-notes-2.4.0]]
== {n} version 2.4.0

[[breaking-2.4.0]]
[float]
=== Breaking changes

* Configure Elastic Agent host path volume to point to correct path {pull}5890[#5890] (issue: {issue}4428[#4428])

> Fleet-managed Elastic Agents now default to use `hostPath` instead of `emptyDir` for storing their state. This will prevent from correctly deploying a second Pod from the same Elastic Agent Deployment on the same Kubernetes node. For cases where this is desired, the volume type can be changed. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-elastic-agent-fleet-known-limitations.html#k8s_storing_local_state_in_host_path_volume_2[docs] to learn more.

Reviewer: ```suggestion
Fleet-managed Elastic Agents now default to use a `hostPath` volume for storing their state. This will prevent more than one Pod from the same Elastic Agent Deployment to be deployed on the same Kubernetes node. For cases where this is desired, the volume type can be changed to an `emptyDir` volume. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-elastic-agent-fleet-known-limitations.html#k8s_storing_local_state_in_host_path_volume_2[docs] to learn more.
```

They were using the container's filesystem before not emptyDir afaik.

[float]
=== Misc

* Update dependency docker.io/library/golang to v1.18.5 {pull}5907[#5907]
* Update k8s to v0.24.3 {pull}5904[#5904]
* Update module sigs.k8s.io/kustomize/kyaml to v0.13.8 {pull}5900[#5900]
* Update module helm.sh/helm/v3 to v3.9.2 {pull}5876[#5876]
* Update dependency golang to v1.18.4 {pull}5873[#5873]
* Update dependency registry.access.redhat.com/ubi8/ubi-minimal to v8.6-854 {pull}5855[#5855]
* Update module sigs.k8s.io/controller-tools to v0.9.1 {pull}5842[#5842]
* Update module github.com/elastic/go-ucfg to v0.8.6 {pull}5841[#5841]
* Update module sigs.k8s.io/controller-runtime to v0.12.2 {pull}5828[#5828]
* Update module github.com/google/go-containerregistry to v0.10.0 {pull}5821[#5821]
* Update module k8s.io/klog/v2 to v2.70.0 {pull}5819[#5819]
* Update module github.com/spf13/cobra to v1.5.0 {pull}5811[#5811]
* Update module github.com/prometheus/common to v0.35.0 {pull}5808[#5808]
* Update module github.com/stretchr/testify to v1.7.3 {pull}5807[#5807]
* Update k8s to v0.24.2 {pull}5792[#5792]
* Update module github.com/hashicorp/vault/api to v1.7.2 {pull}5761[#5761]
* Update module github.com/hashicorp/vault/api to v1.7.1 {pull}5754[#5754]

> * Update module github.com/stretchr/testify to v1.7.2 {pull}5734[#5734]

Reviewer 2: Could you regenerate? I added the `exclude-from-release-notes` label to keep only the last update when there were several for the same dependency:
```diff
* Update k8s to v0.24.3 {pull}5904[#5904]
-* Update k8s to v0.24.2 {pull}5792[#5792]
* Update module github.com/hashicorp/vault/api to v1.7.2 {pull}5761[#5761]
-* Update module github.com/hashicorp/vault/api to v1.7.1 {pull}5754[#5754]
* Update module github.com/stretchr/testify to v1.7.3 {pull}5807[#5807]
-* Update module github.com/stretchr/testify to v1.7.2 {pull}5734[#5734]
```
